pipeline/src/lib.rs

```diff
@@ -34,6 +34,12 @@ impl HostContext {
         (ctx, cb)
     }
 
+    pub fn clear_outputs(&mut self) {
```

Comment:
shouldn't this be private and the pipeline then responsible for calling this when appropriate?

Reply:
but will pipeline be able to call this if it's private?

Reply:
what I was thinking is if this should be "automatic" for pipeline users (as in, isn't it a kind of artifact too, that needs to be cleared if you run some part of the pipeline again)

Reply:
it's kind of part of the witness, but in a different data structure. But I'm not sure I understand what the suggestion is